samples/drivers/dac/src/main.c

```diff
@@ -8,48 +8,32 @@
 #include <sys/printk.h>
 #include <drivers/dac.h>
 
-#if defined(CONFIG_BOARD_NUCLEO_F091RC) || \
-	defined(CONFIG_BOARD_NUCLEO_G071RB) || \
-	defined(CONFIG_BOARD_NUCLEO_G431RB) || \
-	defined(CONFIG_BOARD_NUCLEO_L073RZ) || \
-	defined(CONFIG_BOARD_NUCLEO_L152RE) || \
-	defined(CONFIG_BOARD_NUCLEO_F429ZI) || \
-	defined(CONFIG_BOARD_NUCLEO_F767ZI) || \
-	defined(CONFIG_BOARD_RONOTH_LODEV)
-#define DAC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(dac1))
-#define DAC_CHANNEL_ID		1
-#define DAC_RESOLUTION		12
-#elif defined(CONFIG_BOARD_TWR_KE18F)
-#define DAC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(dac0))
-#define DAC_CHANNEL_ID		0
-#define DAC_RESOLUTION		12
-#elif defined(CONFIG_BOARD_FRDM_K64F)
-#define DAC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(dac0))
-#define DAC_CHANNEL_ID		0
-#define DAC_RESOLUTION		12
-#elif defined(CONFIG_BOARD_FRDM_K22F)
-#define DAC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(dac0))
-#define DAC_CHANNEL_ID		0
-#define DAC_RESOLUTION		12
-#elif defined(CONFIG_BOARD_ARDUINO_ZERO)
-#define DAC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(dac0))
-#define DAC_CHANNEL_ID		0
-#define DAC_RESOLUTION		10
+#define ZEPHYR_USER_NODE DT_PATH(zephyr_user)
```

Comment:
should we provide a pre-defined macro for such node, e.g. `DT_USER_NODE`?

Reply:
Maybe we can leave that to follow-up work? I'm not sure about that since we've always done it this way since inferred bindings for the `/zephyr,user` node were introduced, and I kind of like that this style makes it clear there's "no magic" from a devicetree.h API perspective.

Reply:
sounds good